maltoolbox/language/classes_factory.py

```diff
@@ -184,3 +184,59 @@ def _create_classes(self) -> None:
         # Once we have the JSON schema we create the actual classes.
         builder = pjs.ObjectBuilder(self.json_schema)
         self.ns = builder.build_classes(standardize_names=False)
+
+    def get_association_by_signature(
+        self,
+        assoc_name: str,
+        left_asset: str,
+        right_asset: str
+    ) -> Optional[str]:
+        """
+        Get association name based on its signature. This is primarily
+        relevant for getting the exact association full name when multiple
+        associations with the same name exist.
+
+        Arguments:
+        assoc_name          - the association name
+        left_asset          - the name of the left asset type
+        right_asset         - the name of the right asset type
+
+        Return: The matching association name if a match is found.
+        None if there is no match.
+        """
+        if not assoc_name in self.json_schema['definitions']\
```

Comment:
I would split out self.json_schema['definitions']['LanguageAssociation']['definitions'] into a variable called 'association_definitions' or something, since it is used more than once and increase readability